core/services/feeds/service.go

```diff
@@ -159,10 +167,52 @@ func (s *service) CountManagers() (int64, error) {
 	return s.orm.CountManagers()
 }
 
+// CreateJobProposal creates a job proposal.
 func (s *service) CreateJobProposal(jp *JobProposal) (int64, error) {
 	return s.orm.CreateJobProposal(context.Background(), jp)
 }
 
+// GetJobProposal gets a job proposal by id.
+func (s *service) GetJobProposal(id int64) (*JobProposal, error) {
+	return s.orm.GetJobProposal(context.Background(), id)
+}
+
+func (s *service) RejectJobProposal(id int64) error {
+	if s.fmsClient == nil {
+		return errors.New("fms rpc client is not connected")
+	}
+
+	ctx := context.Background()
+
+	jp, err := s.orm.GetJobProposal(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	if jp.Status != JobProposalStatusPending {
+		return errors.New("must be a pending job proposal")
+	}
+
+	err = s.txm.Transact(func(ctx context.Context) error {
+		if err = s.orm.UpdateJobProposalStatus(ctx, id, JobProposalStatusRejected); err != nil {
+			return err
+		}
+
+		if _, err = s.fmsClient.RejectedJob(context.Background(), &pb.RejectedJobRequest{
```

Comment:
does FMS handle cancellations? If so maybe we do want to set a timeout on this one to bound it

Reply:
wsrpc does not respect context for cancellations but has a default timeout of 2 seconds for the request. 

There is a TODO item in wsrpc to add this feature in though.

Reply:
I've passed in the parent context here so when does wsrpc does support it it will work without changes in the node